central/reports/scheduler/v2/reportgenerator/report_gen_view_based_test.go

```diff
@@ -0,0 +1,638 @@
+//go:build sql_integration
+
+package reportgenerator
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	blobDS "github.com/stackrox/rox/central/blob/datastore"
+	clusterDSMocks "github.com/stackrox/rox/central/cluster/datastore/mocks"
+	"github.com/stackrox/rox/central/graphql/resolvers"
+	"github.com/stackrox/rox/central/graphql/resolvers/loaders"
+	namespaceDSMocks "github.com/stackrox/rox/central/namespace/datastore/mocks"
+	deploymentsView "github.com/stackrox/rox/central/views/deployments"
+	imagesView "github.com/stackrox/rox/central/views/images"
+	watchedImageDS "github.com/stackrox/rox/central/watchedimage/datastore"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	postgresSchema "github.com/stackrox/rox/pkg/postgres/schema"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+func TestViewBasedReporting(t *testing.T) {
+	suite.Run(t, new(ViewBasedReportingTestSuite))
+}
+
+type ViewBasedReportingTestSuite struct {
+	suite.Suite
+	mockCtrl *gomock.Controller
+
+	ctx                   context.Context
+	testDB                *pgtest.TestPostgres
+	reportGenerator       *reportGeneratorImpl
+	resolver              *resolvers.Resolver
+	watchedImageDatastore watchedImageDS.DataStore
+	clusterDatastore      *clusterDSMocks.MockDataStore
+	namespaceDatastore    *namespaceDSMocks.MockDataStore
+	blobStore             blobDS.Datastore
+}
+
+type viewBasedReportData struct {
+	deploymentNames []string
+	imageNames      []string
+	componentNames  []string
+	cveNames        []string
+	cvss            []float64
+}
+
+func (s *ViewBasedReportingTestSuite) SetupSuite() {
+	s.ctx = loaders.WithLoaderContext(sac.WithAllAccess(context.Background()))
+	s.testDB = resolvers.SetupTestPostgresConn(s.T())
+}
+
+func (s *ViewBasedReportingTestSuite) SetupTest() {
+	s.mockCtrl = gomock.NewController(s.T())
+
+	// Create data stores based on feature flag
+	imageDataStore := resolvers.CreateTestImageV2Datastore(s.T(), s.testDB, s.mockCtrl)
```

Comment:
Not for your PR , but we should rename this to `CreateTestImageDatastoreForCVEV2` so that it is not confused with datastore for V2 images.